src/Core/Algorithms/Legacy/Fields/RegisterWithCorrespondences.cc

```diff
@@ -662,25 +741,22 @@ RegisterWithCorrespondencesAlgo::runN(FieldHandle input, FieldHandle Cors1, Fiel
  
   omesh=imesh;
   
-  algo_end(); return (true);
+  return (true);
 }
 
-
-bool
-RegisterWithCorrespondencesAlgo::radial_basis_func(VMesh* Cors, VMesh* points, MatrixHandle& Smat)
+bool RegisterWithCorrespondencesAlgo::radial_basis_func(VMesh* Cors, VMesh* points, MatrixHandle& Smat) const
 {
     VMesh::Node::size_type num_cors, num_pts;
     VMesh::Node::iterator iti,itj;
-    SCIRun::Point Pc,Pp;
+    Point Pc,Pp;
   
     
     Cors->size(num_cors);
     points->size(num_pts);
     
     double xcomp=0.0,ycomp=0.0,zcomp=0.0,mag=0.0;
-    //MatrixHandle SMat;
-    Smat=new DenseMatrix(num_pts,num_cors);
-    DenseMatrix *Sigma = dynamic_cast<DenseMatrix*>(Smat->dense());
+    //Smat = matrix_convert::to_dense(Smat);
+    DenseMatrixHandle Sigma = matrix_convert::to_dense(Smat);
```

Comment:
@mhansen1 Conversion error: original code allocated a dense matrix here, please add it back. Code crashes with null pointer.
```
//original
Smat=new DenseMatrix(num_pts,num_cors);
DenseMatrix *Sigma = dynamic_cast<DenseMatrix*>(Smat->dense());
```